src/toolbar/draftMode/draftModeIntergration.js

```diff
@@ -0,0 +1,13 @@
+import { draftModeServer } from "./draftModeServer";
+
+export default {
+  name: "Draft-Mode",
+  hooks: {
+    "astro:config:setup": ({ addDevToolbarApp }) => {
+      addDevToolbarApp("./src/toolbar/draftMode/draftModeClient.js");
+    },
+    "astro:server:setup": ({ server }) => {
+      draftModeServer(server);
+    },
```

Comment:
Just want to know...
Is it necessary to add a draftModeServer file?
Is there any use case for that?
I just didn't think it was necessary. Plz, correct me if I'm wrong.

Reply:
If you haven't already, I would recommend running it up locally and watching the logs as you toggle it, it might be easier to understand than the explanation I am about to give.

From my understanding, Astro being server side rendered also happens in dev mode, you just get the recompile and browser reload. Therefore we need a way to inject the flag into the compile phase as the draft blog posts will never be parsed due to the filter. 

The Astro server part then has the listener for the web socket to update the flag and store the state. when the browser reloads, astro reruns the filter and now parses the extra files that are marked draft. 

The is the first time I have done anything with Astro or SSR and I am not too interested in how they work under the hood, I used the theme so I could just write my blog but there are some interesting bits to play with. So I just approached it as a simple problem rather than looking deeper